scipy/_lib/_array_api.py

```diff
@@ -163,3 +163,53 @@ def copy(x, *, xp=None):
         xp = array_namespace(x)
 
     return as_xparray(x, copy=True, xp=xp)
+
+
+def assert_equal(actual, desired, err_msg='', xp=None):
+    # TODO compare arrays from different libraries
+    if xp is None:
+        xp = array_namespace(actual)
+    if 'cupy' in xp.__name__:
+        return xp.testing.assert_array_equal(actual, desired, err_msg=err_msg)
+    elif 'torch' in xp.__name__:
+        return xp.testing.assert_close(actual, desired, rtol=0, atol=0,
+                                       msg=err_msg)
```

Comment:
from https://pytorch.org/docs/stable/testing.html:
![image](https://github.com/scipy/scipy/assets/51488791/3a19abcd-3d4c-4297-9d81-7277e427b4a5)

Reply:
Funky but recommended indeed. You may want to add a comment in the line above to avoid the next reader of this code being puzzled. Something like:
```
# PyTorch recommends using `rtol=0, atol=0` like this to test for exact equality
```